utils/libtokencap/make_dict_v2.sh

```diff
@@ -0,0 +1,49 @@
+#default values
+timeout_sec=5
+LD_PRELOAD_PATH="/home/${USER}/AFLplusplus/utils/libtokencap/libtokencap.so"
+
+#help
+usage() {
+    echo "Usage: $0 -o <target_output> -b <target_bin> [-t <timeout_sec>] [-p <LD_PRELOAD_PATH>]"
+    echo "Options:"
+    echo "  -o  Path to target output directory"
+    echo "  -b  Path to target program binary"
+    echo "  -t  Timeout in seconds (default: 5)"
+    echo "  -p  Path to LD_PRELOAD library (default: ${LD_PRELOAD_PATH})"
+    exit 1
+}
+
+#parse cli options
+while getopts ":o:b:t:p:" opt; do
+    case $opt in
+        o) target_output="$OPTARG" ;;
+        b) target_bin="$OPTARG" ;;
+        t) timeout_sec="$OPTARG" ;;
+        p) LD_PRELOAD_PATH="$OPTARG" ;;
+        \?) echo "Invalid option: -$OPTARG" >&2; usage ;;
+        :) echo "Option -$OPTARG requires an args" >&2; usage ;;
+    esac
+done
+
+#check options
+if [ -z "$target_output" ] || [ -z "$target_bin" ]; then
+    echo "Error: Missing mandatory opts" >&2
+    usage
+fi
+
```

Comment:
put a check here that error if both LD_PRELOAD and LD_PRELOAD_PATH are empty.

Reply:
seems yes, its a localized point, worth making more universal.

Reply:
that is not what I meant but thinking about it, what you did is better